bindings/rust/bench/src/openssl.rs

```diff
@@ -94,17 +134,40 @@ impl TlsConnection for OpenSslConnection {
                     builder.set_ca_file(get_cert_path(CACert, crypto_config.sig_type))?;
                     builder.set_verify(SslVerifyMode::FAIL_IF_NO_PEER_CERT | SslVerifyMode::PEER);
                 }
+                if handshake_type == HandshakeType::Resumption {
+                    builder.set_session_cache_mode(SslSessionCacheMode::CLIENT);
+                }
             }
         }
-
-        Ok(builder.build())
+        Ok(Self::Config {
+            config: builder.build(),
+            session_ticket_storage,
+        })
     }
 
     fn new_from_config(
         config: &Self::Config,
         connected_buffer: ConnectedBuffer,
     ) -> Result<Self, Box<dyn Error>> {
-        let connection = SslStream::new(Ssl::new(config)?, connected_buffer.clone())?;
+        // check if there is a session ticket available
+        // a session ticket will only be available if the Config was created
+        // with session resumption enabled
+        let maybe_ticket = config
+            .session_ticket_storage
+            .stored_ticket
+            .lock()
+            .unwrap()
+            .take();
+        if let Some(ticket) = &maybe_ticket {
+            let _result = unsafe { config.config.add_session(ticket) };
+        }
+
+        let mut connection = Ssl::new(&config.config)?;
+        if let Some(ticket) = &maybe_ticket {
```

Comment:
Seems kind of strange that you have to add the ticket to both the config and the connection?

Reply:
The problem is that the ticket needs to be stored _somewhere_. I guess you could have the handshake return it at some point? But rustls already sort of made that decision for us through their SessionCache api: https://doc.servo.org/rustls/client/trait.ClientSessionStore.html

In rustls you don't even have to manually set the session ticket on a connection, it's automatically done by SNI. The current implementation for s2n-tls and openssl just seeks to match that layer of abstraction.